nav2_regulated_pure_pursuit_controller/include/nav2_regulated_pure_pursuit_controller/regulated_pure_pursuit_controller.hpp

```diff
@@ -244,20 +245,118 @@ class RegulatedPurePursuitController : public nav2_core::Controller
     const geometry_msgs::msg::Point & p2,
     double r);
 
+  /**
+    * @brief normalize
+    * Normalizes the angle to be -M_PI circle to +M_PI circle
+    * It takes and returns radians.
+    * @param angles Angles to normalize
+    * @return normalized angles
+    */
+  template<typename T>
+  auto normalize_angles(const T & angles)
+  {
+    auto && theta = fmod(angles + M_PI, 2.0 * M_PI);
+    if (theta < 0)
+      theta += 360;
+    return theta - M_PI;
+  }
+
+
+  /**
+    * @brief shortest_angular_distance
+    *
+    * Given 2 angles, this returns the shortest angular
+    * difference.  The inputs and ouputs are of course radians.
+    *
+    * The result
+    * would always be -pi <= result <= pi.  Adding the result
+    * to "from" will always get you an equivelent angle to "to".
+    * @param from Start angle
+    * @param to End angle
+    * @return Shortest distance between angles
+    */
+  template<typename F, typename T>
+  auto shortest_angular_distance(
```

Comment:
I believe both of these functions can be used from `angles` rather than reimplemented https://github.com/ros/angles

Reply:
I suppose so. I follow MPPI controller utils.hpp implementation in humble. I see Nav2 Rolling switched to `angles` dependency